### PR TITLE
Mod backup moodle2

### PR DESCRIPTION
**Pull request by alexandru-elisei**

### CODE REVIEW

**Comment by alexandru-elisei** [comment]
I think the Travis error is wrong, PHPUnit function names use camel case.

File: classes/local/util/manager.php [hunk 1]
```diff
@@ -289,27 +289,45 @@ protected function prepare_mod_files() {
...
         }
+    }
 
-        if (!$this->should_have('upgrade')) {
-            $this->logger->warning('Upgrade feature not defined');
-        }
+    /*
+     * Prepares the skeleton files for the 'backup_moodle2' feature for an activity module.
+     */
+    public function prepare_mod_backup_moodle2() {
```
Comment (alexandru-elisei) [inline classes/local/util/manager.php]: According to the [Backup API](https://docs.moodle.org/dev/Backup_API) each plugin type implements the backup feature differently: different file names, different files, different file content.

I chose to have a function like this for each plugin type that implements the backup feature. So there'll be:
- function prepare_mod_backup_moodle2() to prepare the backup files for activity modules
- function prepare_block_backup_moodle2() to prepare the backup file for blocks
- function prepare_themes_backup_moodle2() to prepare the backup for themes
- function prepare_report_backup_moodle2() to prepare the backup for reports. I've found no documentation about this, and no report plugin that implements this feature. I think we'll leave this out for now.

**Comment by mudrd8mz** [comment]
I reported the Travis false error here: https://github.com/moodlerooms/moodle-plugin-ci/issues/26

**Comment by alexandru-elisei** [comment]
If you're ok with how I've implemented the backup feature, I'll implement it the same way for blocks and local/tool plugins, so we can have four plugin types implemented completely.

File: classes/local/util/manager.php (hunk 1 above)
Comment (mudrd8mz) [inline classes/local/util/manager.php]: > I think we'll leave this out for now.

+1

**Pull request merged**